velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -497,12 +497,17 @@ uint32_t HiveDataSink::appendWriter(const HiveWriterId& id) {
   options.compressionKind = insertTableHandle_->compressionKind();
   options.setMemoryReclaimer = connectorQueryCtx_->setMemoryReclaimer();
   ioStats_.emplace_back(std::make_shared<dwio::common::IoStatistics>());
+  // TODO: we need to set memory reclaimer of the created leaf memory pool to
```

Comment:
This will break Meta internal use case. Why we can't use connectorQueryCtx_->memoryPool() for this? Thanks!

Reply:
The file writer uses the connector memory pool. I felt the file sink for S3 should use the connector memory pool as well. But I see that file sink could be memory in which case the current operator memory pool is better. I will close this for now.